Add an explenation on the advisor section (#1213)
scylladb/scylla-monitoring: docs/source/advisor.rst

=========================
Scylla Monitoring Advisor
=========================

> The Advisor is both a concept and an implementation.

Reviewer: This is just confusing.
the state of Israel is also concept and an implementation

I suggest something like:

Scylla Advisor is an element of Scylla Monitoring which recognize bad practice usage and advice on how to fix it

> The advisor concept

Reviewer 2: ```suggestion
The Advisor Concept
```

^^^^^^^^^^^^^^^^^^^^^

> The advisor looks for potential problems with the system,  with the setup, or with the data model and notify about it. The idea is to give insight into the information rather than general graphs.

Reviewer 2: ```suggestion
The Advisor looks for potential problems with the system,  with the setup, or with the data model and notifies about it. The idea is to give additional insight that the general graphs cannot provide.
```

> The advisor looks for potential problems with the system,  with the setup, or with the data model and notify about it. The idea is to give insight into the information rather than general graphs.

Reviewer: ```suggestion
The advisor looks for potential problems with the system,  with the setup, or with the data model and notify about it. 
```

Reviewer: We should not explain ideas, rather what is there now

> This approach is used in the `CQL optimization section` and the advisor section.

Reviewer 2: ```suggestion
This approach is used in the `CQL optimization section` and the Advisor section.
```

.. _`CQL optimization section`: ./cql_optimization

> The advisor section

Reviewer 2: ```suggestion
The Advisor Section
```

^^^^^^^^^^^^^^^^^^^^
.. figure:: advisor_panel.png

    **The Advisor section**

> The advisor section is on the Overview dashboard and consists of two parts.

Reviewer 2: ```suggestion
The Advisor section is on the Overview dashboard and consists of two parts.
```

> On the left, there is a table with issues found by the advisor.

Reviewer 2: ```suggestion
On the left, there is a table with issues found by the Advisor.
```

Each issue has a category, a dashboard quick link to jump to the relevant dashboard, and a description of the issue.

> For example, the advisor would warn if there is a large cell in the data. A few large cells usually indicate a problem with the data model or a problem with the client code. This issue can have an impact on overall system performance.

Reviewer 2: ```suggestion
For example, the Advisor would warn if there is a large cell in the data. A few large cells usually indicate a problem with the data model or a problem with the client code. This issue can have an impact on overall system performance.
```

> On the right side, there is the system balance section. This section looks for outliers shards or nodes. If one shard behaves very differently from other shards, it indicates a problem with the system.

Reviewer 2: ```suggestion
The System Balance section is on the right. This section looks for outlier shards or nodes. If one shard behaves very differently from other shards, it indicates a problem with the system.
```

> The advisor section is on the Overview dashboard and consists of two parts.
> On the left, there is a table with issues found by the advisor.
> Each issue has a category, a dashboard quick link to jump to the relevant dashboard, and a description of the issue.
> For example, the advisor would warn if there is a large cell in the data. A few large cells usually indicate a problem with the data model or a problem with the client code. This issue can have an impact on overall system performance.
> 
> On the right side, there is the system balance section. This section looks for outliers shards or nodes. If one shard behaves very differently from other shards, it indicates a problem with the system.
> For example, in the case of hot-partition, where a single partition gets many requests and acts as a bottleneck, the balance section will indicate that the latency and cache hits differ between shards. 

Reviewer: ```suggestion
The advisor section is on the Overview dashboard and consists of two parts:

On the left,  the advisor issues table.
Each issue has a category, a link to jump to a relevant dashboard, and a description of the issue.
For example, the advisor would warn of large cells in the data, usually indicating a problem with the data model or a problem with the client code, impacting system performance.

On the right, the system balance section. 
This section notifies for an imbalance between shards or nodes, indicating a potential problem.
For example, when a single, hot partition gets most of the requests, making one shard a bottleneck, the balance section will indicate that the latency and cache hits are imbalanced between shards.
```

> For example, in the case of hot-partition, where a single partition gets many requests and acts as a bottleneck, the balance section will indicate that the latency and cache hits differ between shards. 

Reviewer 2: ```suggestion
For example, in the case of a hot-partition, where a single partition gets many requests and acts as a bottleneck, the Balance section will indicate that the latency and cache hits differ between shards. 
```
